Add contributor license agreement (#618)
comit-network/comit-rs: .github/PULL_REQUEST_TEMPLATE.md

[Describe your change]
Got some code for us? Awesome 🎊!

Please include a description of your change & check your PR against this list, thanks!

- [ ] If it resolves a ticket, please add `Resolves #[Issue number]`
- [ ] Commit messages are meaningful
- [ ] The code is formatted using `cargo fmt`

> - [ ] The project compiles: `cargo make build`

Reviewer: The project and tests runs: `cargo make all`

> - [ ] The project compiles: `cargo make build`

Author: ```suggestion
- [ ] The project compiles: `cargo make all`
```

> - [ ] The project and tests runs: `cargo make all`

Reviewer: ```suggestion
- [ ] The project builds and tests run: `cargo make all`
```

> - [ ] You agree to the [Contributor License Agreement](../CONTRIBUTING.md)

Reviewer: ❌ This does not point to the CLA

Author: CLA is in `CONTRIBUTING.md`... I wanted to link to that section directly (e.g. (`../CONTRIBUTING.md#CLA`), but this does not work :(

> - [ ] You agree to the [Contributor License Agreement](../CONTRIBUTING.md)

Reviewer: ```suggestion
- [ ] You agree to the [Contributor License Agreement](../CONTRIBUTING.md#contributor-license-agreement)
```
